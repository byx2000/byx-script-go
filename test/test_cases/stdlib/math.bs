import math

println(Math.abs(15))
printf('%.3f\n', Math.abs(-3.14))
printf('%.3f\n', Math.sin(10))
printf('%.3f\n', Math.sin(12.34))
printf('%.3f\n', Math.cos(10))
printf('%.3f\n', Math.cos(12.34))
printf('%.3f\n', Math.tan(10))
printf('%.3f\n', Math.tan(12.34))
println(Math.pow(2, 3))
printf('%.3f\n', Math.pow(2, 3.5))
printf('%.3f\n', Math.pow(2.5, 3))
printf('%.3f\n', Math.pow(2.5, 3.5))
printf('%.3f\n', Math.exp(2))
printf('%.3f\n', Math.exp(3.14))
printf('%.3f\n', Math.ln(25))
printf('%.3f\n', Math.ln(12.56))
printf('%.3f\n', Math.log10(25))
printf('%.3f\n', Math.log10(12.56))
printf('%.3f\n', Math.sqrt(2))
printf('%.3f\n', Math.sqrt(31.5))
printf('%.3f\n', Math.round(7))
printf('%.3f\n', Math.round(8.3))
printf('%.3f\n', Math.round(12.9))
printf('%.3f\n', Math.ceil(7))
printf('%.3f\n', Math.ceil(8.3))
printf('%.3f\n', Math.ceil(12.9))
printf('%.3f\n', Math.floor(7))
printf('%.3f\n', Math.floor(8.3))
printf('%.3f\n', Math.floor(12.9))
