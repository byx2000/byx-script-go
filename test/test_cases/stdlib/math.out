15
3.140
-0.544
-0.224
-0.839
0.974
0.648
-0.230
8
11.314
15.625
24.705
7.389
23.104
3.219
2.531
1.398
1.099
1.414
5.612
7.000
8.000
13.000
7.000
9.000
13.000
7.000
8.000
12.000